### PR TITLE
Fix layout and button click for iOS Counter sample

### PR DESCRIPTION
**Pull request by DylanSale**
Fix? for #930 and #903 
Screenshot 2023-04-21 at 9 19 31 pm

Specifically there are 2 main issues 
1. It seems that internal classes don't get Obj-c classes generated, so they can't be added to views or used as selectors for messages (ie, clicked). So I made the `FlexContainerHostView` and `IosButton` classes public for those reasons.
2. The UIStackView that contained the FlexContainerHostView has NSLayoutConstraints enabled, so the layoutSubviews needs to set the height constraints (or width depending on the direction).

I haven't done iOS dev in about 4 years so please review with caution. :)

Another alternative to implementing the constraints would have been to change the `CounterViewController`'s `container.alignment` to `.fill` which also disables layout constraints and just sets the height to the UIStackView's height. I guess the best alternative depends on the use case. 

There is another issue where the Counter text get's ellipsized when it's too wide, I'm not sure what's causing that though.

### CODE REVIEW

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainer.kt [hunk 1]
```diff
@@ -110,7 +107,26 @@ internal class UIViewFlexContainer(
...
   }
 
-  private inner class FlexContainerHostView : UIScrollView(cValue { CGRectZero }) {
+  public inner class FlexContainerHostView : UIScrollView(cValue { CGRectZero }) {
+
+    private var widthContraint: NSLayoutConstraint? = null
+    private var heightConstraint: NSLayoutConstraint? = null
+
+    override fun didMoveToSuperview() {
+      if (superview == null) return
```
Comment (JakeWharton) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainer.kt]: Is it safe not to call super in this case?

I would prefer if we switched to
```kotlin
if (superview != null) {
  // do stuff
}
super.didMoveToSuperview()
```
just to be safe

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainer.kt [hunk 2]
```diff
@@ -110,7 +107,26 @@ internal class UIViewFlexContainer(
...
+      translatesAutoresizingMaskIntoConstraints = false
+      widthContraint = widthAnchor.constraintEqualToConstant(0.0)
+      heightConstraint = heightAnchor.constraintEqualToConstant(0.0)
+      if(direction.isHorizontal) {
+        NSLayoutConstraint.activateConstraints(listOf(widthContraint))
+      } else if (direction.isVertical) {
+        NSLayoutConstraint.activateConstraints(listOf(heightConstraint))
+      } else {
+        //Unknown??
```
Comment (JakeWharton) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainer.kt]:
```suggestion
      throw AssertionError()
```

**Review by JakeWharton — APPROVED** [review]
Please run `spotApply` task to fix a few formatting things (like wildcard imports).

**Comment by DylanSale** [comment]
Sorry for all the pushes, I keep forgetting not to push the cocopods stuff :)

It passes spotApply now.

**Comment by DylanSale** [comment]
I just asked @TwoLivesLeft to look at this and they pointed out that in this case there's no point in using a `UIScrollView` if you are just setting the scrollview's size to the content size because then it will never actually scroll. It only scrolls if the content size is larger than the scrollview's bounds (my interpretation of their words).

I'm not sure what the expected behaviour here should be in that case but I thought I'd raise it as a potential blocker to being merged.

**Comment by JakeWharton** [comment]
Yes the layout implementation is general-purpose and may be used for scrolling views. Today we do not switch the base view type even if you disable scrolling.

**Comment by DylanSale** [comment]
After looking into it more, I think this needs a rework. Nesting Rows and Columns doesn't work at all with the way this is set up. I'll close this for now and make a new PR if I figure out the best approach. Thanks 😄

**Pull request closed without merging**